### PR TITLE
[KernelGen][Nvidia] Add hspmm operator with Triton kernel

### PR DESCRIPTION
**Pull request by ShawnsYing**
## Summary
Adds a Triton kernel for `hspmm`. Performs hybrid sparse-dense matrix multiplication where mat1 is sparse COO and mat2 is dense.

## Testing
- Validated against reference on device via dense matmul
- 292 accuracy tests pass (various M/K/N/nnz combinations)
- Edge cases: empty nnz=0, uncoalesced input, non-contiguous mat2
- Tested on: Nvidia

## Performance
Test command: `pytest benchmark/test_hspmm.py --level core` (NVIDIA H20)

### hspmm

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup | TFLOPS |
|-------|------|--------------------|-------------------|---------|--------|
| float32 | [384, 384] | 0.049600 | 0.008800 | 5.636 | 0.000 |
| float32 | [4096, 4096] | 0.049536 | 0.008480 | 5.842 | 0.002 |
| float32 | [1024, 1024] | 0.049504 | 0.008000 | 6.188 | 0.004 |
| float32 | [2048, 2048] | 0.049888 | 0.008256 | 6.043 | 0.008 |
| float32 | [4096, 4096] | 0.050528 | 0.008736 | 5.784 | 0.015 |
| float64 | [384, 384] | 0.052480 | 0.008208 | 6.394 | 0.000 |
| float64 | [4096, 4096] | 0.070400 | 0.009728 | 7.237 | 0.002 |
| float64 | [1024, 1024] | 0.051840 | 0.008896 | 5.827 | 0.004 |
| float64 | [2048, 2048] | 0.058368 | 0.008576 | 6.806 | 0.008 |
| float64 | [4096, 4096] | 0.072256 | 0.009184 | 7.868 | 0.014 |

| Operator | Geometric Mean Speedup |
|----------|------------------------|
| hspmm | **6.299** |

## Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (H20) | PASS (292 cases) | 6.299 | Primary |
| Tianshu | N/A | — | — |
| Muxi | N/A | — | — |
| Ascend | N/A | — | — |
| Hygon | N/A | — | — |

## Files Changed
- `src/flag_gems/ops/hspmm.py`: Triton kernel implementation with SparseCUDA dispatch
- `tests/test_hspmm.py`: Accuracy test
- `benchmark/test_hspmm.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG` with SparseCUDA dispatch key
- `conf/operators.yaml`: Add operator entry (kind: BLAS, stage: alpha 5.4)

## Notes
- Only float32/float64 supported (CUDA aten::hspmm limitation on sparse ops)
- Uses SparseCUDA dispatch key (4-tuple registration in _FULL_CONFIG)
- Output is uncoalesced sparse COO (matching CUDA behavior)

### CODE REVIEW

File: benchmark/test_hspmm.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import pytest
```
Comment (bin913) [inline benchmark/test_hspmm.py:1]: Please add copyright and license

File: tests/test_hspmm.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import pytest
```
Comment (bin913) [inline tests/test_hspmm.py:1]: Please add copyright and license

**Comment by ShawnsYing** [comment]
Addressed both comments — the two files were indeed the only ones in the PR missing the header. Rebuilt on current master (`4bea493`, 6 files, additions only; the branch had also drifted into a conflict with master, now resolved).

- `tests/test_hspmm.py` — Apache-2.0 header added.
- `benchmark/test_hspmm.py` — Apache-2.0 header added.
- `src/flag_gems/ops/hspmm.py` already had it.

No other changes to the operator. Also moved the `operators.yaml` entry to `stages: - alpha: '5.5.0'`.

**Verified:** 292 passed in `tests/test_hspmm.py`; black/isort/flake8 clean; `sort_exports`, `check_init_exports`, `check_operators_yaml`, `check_operator_markers`, `check_kernelgen_tests`, `check_benchmark_coverage`, `check_aten_operators`, `check_api_logs` all pass.

**Review by bin913 — APPROVED** [review]
